pip/commands/download.py

```diff
@@ -196,7 +196,8 @@ def run(self, options, args):
                     None
                 )
 
-                requirement_set.prepare_files(finder)
+                resolver = self._build_resolver(options, finder)
+                resolver.resolve(requirement_set)
```

Comment:
It appears like the way this is working is to mutate the ``RequirementSet`` that we're passing in. That is OK for this PR since that more closely resembles the existing code (and thus reduces the amount of code churn), but thinking forward I think it would be great if we can get this code refactored so we're passing in the things we want to install, and then it returns a new thing that lists everything we need to install. These could both be ``RequirementSets``s possibly?

Anyways, not an immediate concern, just something to keep in mind.